include/internal/refcount.h

```diff
@@ -63,6 +63,26 @@ static ossl_inline int CRYPTO_DOWN_REF(int *val, int *ret, void *lock)
     return 1;
 }
 
+# elif defined(_MSC_VER) && _MSC_VER>=1200
+
+#  define HAVE_ATOMICS 1
+
+#  pragma intrinsic(_InterlockedExchangeAdd)
```

Comment:
Do we need an #include <intrin.h>?

Reply:
intrin.h actually predates _InterlockedExchangeAdd. I.e. old enough MSC that recognizes this pragma doesn't have intrin.h. (At the same time there can be e.g. xmmintrin.h.) Also note that appveyor is all green. Or rather more specifically 32-bit build which *fails* on warnings. I can confirm warning-free compilation with multiple versions. Well, one can argue that we don't support such old version, identify first that has intrin.h and adjust _MSC_VER>xx00 accordingly...